Fix KVRole type annotation

hl475 opened the pull request:
Summary:
`KVRole = Literal[KVProducer, KVConsumer]` doesn’t mean “union of those literals.” but nests literals inside a literal. This will let some pyre/mypy/pyright complain from the downstream.

This diff also fix lint issues from the same file about trailing spaces

Test Plan:
Exist CI

Rollback Plan:

Differential Revision: D81659853

houseroad commented on vllm/config/__init__.py:
```diff
@@ -3211,7 +3211,7 @@ def _parse_collect_detailed_traces(self):
 KVProducer = Literal["kv_producer", "kv_both"]
 KVConsumer = Literal["kv_consumer", "kv_both"]
-KVRole = Literal[KVProducer, KVConsumer]
+KVRole = Literal["kv_producer", "kv_consumer", "kv_both"]
```
@hl475 , this suggestion makes sense.

22quinn reviewed (approved):
Thanks for the fix!

hl475 commented:
change the PR to draft as the current version breaks Meta downstream build

hl475 committed:
Fix KVRole type annotation (#24415)

Summary:
Pull Request resolved: https://github.com/vllm-project/vllm/pull/24415

`KVRole = Literal[KVProducer, KVConsumer]` doesn’t mean “union of those literals.” but nests literals inside a literal. This will let some pyre/mypy/pyright complain from the downstream.

This diff also fix line issues from the same file about trailing spaces

Test Plan:
Exist CI

Rollback Plan:

Reviewed By: 22quinn, houseroad

Differential Revision: D81659853

hmellor reviewed (changes requested):
The following is correct:

```python
x = Literal["x"]
y = Literal["y"]

z: Literal[x, y]
```

If you are seeing issues with tooling, it is an issue with the tooling, not the typing.

See https://typing.python.org/en/latest/spec/literal.html#legal-and-illegal-parameterizations:

> `Literal` may also be parameterized by other literal types, or type aliases to other literal types. For example, the following is legal:
>
> ```python
> ReadOnlyMode         = Literal["r", "r+"]
> WriteAndTruncateMode = Literal["w", "w+", "wt", "w+t"]
> WriteNoTruncateMode  = Literal["r+", "r+t"]
> AppendMode           = Literal["a", "a+", "at", "a+t"]
>
> AllModes = Literal[ReadOnlyMode, WriteAndTruncateMode,
>                   WriteNoTruncateMode, AppendMode]
> ```
>
> This feature is again intended to help make using and reusing literal types more ergonomic.
>
> **Note**: As a consequence of the above rules, type checkers are also expected to support types that look like the following:
>
> ```python
> Literal[Literal[Literal[1, 2, 3], "foo"], 5, None]
> ```
>
> This should be exactly equivalent to the following type:
>
> ```python
> Literal[1, 2, 3, "foo", 5, None]
> ```

hmellor commented:
```python
>>> from typing import Literal
>>> x = Literal["x"]
>>> y = Literal["y"]
>>> Literal[x, y]
typing.Literal['x', 'y']
```
